src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/Builder.cs

```diff
@@ -6,15 +6,31 @@
 using System.Text;
 using System.Collections.Generic;
 using System.Text.RegularExpressions;
+using System.Threading;
 
 namespace Xamarin.ProjectTools
 {
 	public class Builder : IDisposable
 	{
+		string buildLogFullPath;
 		public bool IsUnix { get; set; }
 		public bool RunningMSBuild { get; set; }
 		public LoggerVerbosity Verbosity { get; set; }
-		public string LastBuildOutput { get; set; }
+		public IEnumerable<string> LastBuildOutput {
+			get {
+				if (!string.IsNullOrEmpty (buildLogFullPath) && File.Exists (buildLogFullPath)) {
+					using (var fs = File.OpenRead (buildLogFullPath)) {
+						using (var sr = new StreamReader (fs, Encoding.UTF8, true, 65536)) {
+							string line;
+							while ((line = sr.ReadLine ()) != null) {
+								yield return line;
```

Comment:
One word of caution here: this implementation means that `LastBuildOutput` can only be enumerated *once*. Anything which involves multiple enumeration will fail.

Additionally, this doesn't appear to meaningfully differ from [`File.ReadLines()`](https://msdn.microsoft.com/en-us/library/dd383503(v=vs.110).aspx).

Returning to the "multiple enumeration" problem, consider this `csharp` fragment:

```
csharp> using System.IO;
csharp> var e = File.ReadLines("Configuration.props");
csharp> e.Any();                                                                                                                                                                                 true                                                                                                                                                                                             
csharp> e;
{ System.ObjectDisposedException: Cannot read from a closed TextReader.
  at System.IO.StreamReader.ReadLine () [0x00008] in <4fdc5ed61a074cafb49fa42deb20d521>:0 
  at System.IO.File+<ReadLines>d__58.MoveNext () [0x0004e] in <4fdc5ed61a074cafb49fa42deb20d521>:0 
  at Mono.CSharpShell.PrettyPrint (System.IO.TextWriter output, System.Object result) [0x001c4] in <c4aa37c9a68943de840f06a1a5d7109e>:0 
  at Mono.CSharpShell.Evaluate (System.String input) [0x0001a] in <c4aa37c9a68943de840f06a1a5d7109e>:0 
```

`e.Any()` triggers an enumeration. `e` *also* triggers an enumeration, resulting in the `ObjectDisposedException`.

Reply:
...though review of the rest of the PR shows that you've avoided the multiple evaluation problem, as you never store `b.LastBuildOutput` into a reusable variable...

Reply:
I need to update the commit message. The Goal of this PR is to reduce Memory pressure by streaming in the data rather than loading it all at once. `File.ReadLines()` will read all the data into a `string[]` and will use a ton of memory. We hit a few problems with this on windows. So the idea is to search the file line by line rather than loading all of that into memory at once.

Reply:
> `File.ReadLines()` will read all the data into a `string[]` and will use a ton of memory.

No, it won't. `File.ReadAllLines()` will do as you say, but `File.ReadLines()` is a streaming API. From the remarks:

> The ReadLines and ReadAllLines methods differ as follows: When you use ReadLines, you can start enumerating the collection of strings before the whole collection is returned; when you use ReadAllLines, you must wait for the whole array of strings be returned before you can access the array. Therefore, when you are working with very large files, ReadLines can be more efficient.